packages/core/middleware/src/helpers/getAgnosticStatusCode.ts

```diff
@@ -1,14 +1,52 @@
 const getAgnosticStatusCode = (obj: unknown, ...searchedKeys: string[]) => {
-  if (!obj || typeof obj !== 'object') return;
-  for (const key of Object.keys(obj)) {
-    if (searchedKeys.includes(key)) {
-      return obj[key];
+  const getAxiosStatus = () => {
+    if ((obj as any)?.isAxiosError) {
+      return (obj as any)?.response?.status;
     }
-    const statusCode = getAgnosticStatusCode(obj[key], ...searchedKeys);
-    if (statusCode) {
-      return statusCode;
+  };
+
+  const getApolloStatus = () => {
+    if ((obj as any)?.networkError) {
+      return 500;
+    }
+    return typeof (obj as any)?.code === 'string' ? 400 : (obj as any)?.code;
+  };
+
+  const findKey = (currentObj: unknown, searchedKeys: string[], depth = 1) => {
+    if (!currentObj || typeof currentObj !== 'object') return;
+
+    for (const key of Object.keys(currentObj)) {
+      if (searchedKeys.includes(key)) {
+        return currentObj[key];
+      }
+
+      if (depth > 3) return;
+
+      const nextDepth = depth + 1;
+      const statusCode = findKey(currentObj[key], searchedKeys, nextDepth);
+
+      if (statusCode) {
+        return statusCode;
+      }
     }
-  }
+  };
+
+  const getCode = () => {
+    const axiosStatus = getAxiosStatus();
+    const apolloStatus = getApolloStatus();
+
+    if (axiosStatus) {
+      return axiosStatus;
+    }
+
+    if (apolloStatus) {
+      return apolloStatus;
+    }
+
+    return findKey(obj, [...searchedKeys]) || 500;
+  };
+
+  return getCode();
```

Comment:
Why is so complicated? and there is no proper TS support?

how about this:


```ts
const STATUS_FIELDS = ['status', 'status2'] as const

type Axios = {
  isAxiosError: boolean;
  response: {
    satus: number
  }
}
type Apollo = {
  networkError: number
  code: string | number
}

type Status = typeof STATUS_FIELDS[number]

type StatusCode = number | null

type UnknowError = { [K in Status]: number; } & { [x: string]: UnknowError }

const reduceStatus = (narrowObject: UnknowError) => (p: StatusCode, c: string): StatusCode => {
  if (STATUS_FIELDS.includes(c as Status)) {
    return narrowObject[c as Status]
  }

  return p || obtainStatusCode(narrowObject[c])
}

const obtainStatusCode = (givenObject: UnknowError): StatusCode => {
  const obj = givenObject || {}

  return Object.keys(obj)
    .reduce(reduceStatus(obj), null) as unknown as number
}

const getStatusCode = (error: Axios | Apollo | UnknowError): number => {
  if ('isAxiosError' in error) {
    return (error as Axios).response.satus
  }

  if ('networkError' in error) {
    const e: Apollo = error as Apollo

    return typeof e.code === 'string' ? 400 : e.code
  }

  return obtainStatusCode(error) || 200
}
```

Of course would be nice to reveal the right axsios ant apollo error type

Reply:
think how to pass depth param as well : )